### PR TITLE
Do not load HPO-associated genes when loading OMIM terms

### LINKED ISSUE #4151: Huge gene counts for diseases
When you update diseases with scout >= 4.72

image

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #4151

Genes associated to disease are included in the min2genes.txt file, a file which we download from the HPO but we never use. It has likely more specific genes instead of the huge list obtained from the HPO terms

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Load diseases using the latest files and make sure that the disease page contains the expected info

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR, DN
- [x] tests executed by CR

### CODE REVIEW

**Comment by dnil** [comment]
Hm, I see no disease associated genes after 
```
scout update diseases -f .
```
Screenshot 2023-10-20 at 14 26 21
Also fewer diseases??

**Comment by northwestwitch** [comment]
I think that it shows what is found in min2genes, I've checked a few and those with no genes don't have genes in the file. The file looks like this in fact:

image

**Comment by northwestwitch** [comment]
Discussing with @dnil I've realised that is probably genemap2 the file to use to do that, and that mim2genes perhaps was used in the past but has changed. We should remove it and the code parsing it!

**Comment by northwestwitch** [comment]
This is ready for review now

File: scout/load/disease.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 from scout.build.disease import build_disease_term
 from scout.parse.hpo_mappings import parse_hpo_annotations, parse_hpo_to_genes
 from scout.parse.omim import get_mim_phenotypes
-from scout.utils.scout_requests import fetch_hpo_disease_annotation, fetch_hpo_to_genes_to_disease
+from scout.utils.scout_requests import fetch_hpo_disease_annotation
```
Comment (dnil) [inline scout/load/disease.py:11]: 👍

File: scout/server/blueprints/diagnoses/templates/diagnoses/omim_terms.html [hunk 2]
```diff
@@ -48,8 +47,13 @@
                       -
                     {% endfor %}
                   </td><!--end of inheritance-->
-                  <td><span class="text-body">{{ omim.genes|length }}</span></td>
-                  <td><span class="text-body">{{ omim.hpo_terms|length }}</span></td>
+                  <td><span class="text-body">
+                    {% for term in omim.hpo_terms %}
+                      <a class="badge bg-secondary text-white" href="https://hpo.jax.org/app/browse/term/{{term}}" target="_blank" rel="noopener"  >{{term}}</a>
```
Comment (dnil) [inline scout/server/blueprints/diagnoses/templates/diagnoses/omim_terms.html:52]: 💎

**Review by dnil — COMMENTED** [review]
Looks safe and simplifies things a bunch.

These "debug" warnings will stay with prod update:
`DEBUG The following gene symbols could not be found in database: {None}`
It's not essential, but it would make sense to either prune empty gene lists on `disease_term`, much like we do with other Pydantic-introduced empties. Or add a more appropriate `None`/empty-value check in `build_disease_term()`, around here https://github.com/Clinical-Genomics/scout/blob/e4a90c795521ef27c91597f462cedf9ee33e4f9d/scout/build/disease.py#L26.

**Comment by northwestwitch** [comment]
> Looks safe and simplifies things a bunch.
> 
> These "debug" warnings will stay with prod update: `DEBUG The following gene symbols could not be found in database: {None}` It's not essential, but it would make sense to either prune empty gene lists on `disease_term`, much like we do with other Pydantic-introduced empties. Or add a more appropriate `None`/empty-value check in `build_disease_term()`, around here

I remember seeing those None before but I can't reproduce it any more? Tested both by updating  diseases from the latest files downloaded from OMIM and by pointing to the files in scout/demo/resources

**Comment by dnil** [comment]
> > Looks safe and simplifies things a bunch.
> > These "debug" warnings will stay with prod update: `DEBUG The following gene symbols could not be found in database: {None}` It's not essential, but it would make sense to either prune empty gene lists on `disease_term`, much like we do with other Pydantic-introduced empties. Or add a more appropriate `None`/empty-value check in `build_disease_term()`, around here
> 
> I remember seeing those None before but I can't reproduce it any more? Tested both by updating diseases from the latest files downloaded from OMIM and by pointing to the files in scout/demo/resources

Well, all you need is to load with a release_4.72, then go from there. Possibly via downgrade to 4.71, which is what we will have on  `prod`. That just happened to also be the state of one of my locals.

**Comment by northwestwitch** [comment]
> Well, all you need is to load with a release_4.72, then go from there. Possibly via downgrade to 4.71, which is what we will have on prod. That just happened to also be the state of one of my locals.

Switched to branch version_4.72 and loaded HPO/genes and disease, I see no Nones: (which is weird, I remember they were there)

image

Switched to branch version_4.71 and loaded HPO terms and OMIM terms, I don't see any Nones here as well:

image

Switched to this branch and loaded OMIM terms: no Nones 🤔 

image

File: scout/build/disease.py [hunk 3]
```diff
@@ -24,6 +24,8 @@ def build_disease_term(disease_info: dict, alias_genes: dict = {}) -> dict:
     hgnc_symbols_not_found = set()
     hgnc_ids = set()
     for hgnc_symbol in disease_info.get("hgnc_symbols", []):
+        if hgnc_symbol is None:
```
Comment (dnil) [inline scout/build/disease.py:27]:
```suggestion
        if None in hgnc_symbol:
```

**Comment by northwestwitch** [comment]
How the diseases pages look after loading diseases using this branch:

image

**Comment by dnil** [comment]
Nah, still the same:
Screenshot 2023-10-23 at 11 54 11

Thing is the value is not a `None`, I think it might be its a set with a `None`, but you're the Pydantic guru here!

**Comment by northwestwitch** [comment]
> Thing is the value is not a `None`, I think it might be its a set with a `None`, but you're the Pydantic guru here!

The Pydantic validation hasn't even happened yet, That message is printed at line 41 of build_disease_term. Pydantic gets called at line 47.. I'm very confused now

**Comment by northwestwitch** [comment]
Ok, I used branch 4.72 and I see no Nones

image

But at least I can reproduce the issue (the huge amount of genes):

image

<hr>

Then switch to 4.71 (after downgrading pydantic), and reload diseases, still no Nones

image

Database looks as expected, with no HPO genes (before the infamous bug fix that led to the memory sucking issue)

image

<ht>

Using this branch:

image

Database analysis:

image

File: scout/build/disease.py (hunk 3 above)
Comment (northwestwitch) [reply scout/build/disease.py:27]: That line should work, could you please check again? Before this change in the branch  I saw:

image

After the change:

image

**Review by dnil — APPROVED** [review]
Retesting with 4.72 -> pydantic -> update diseases -> 4.71 -> pydantic -> update diseases -> this branch -> pydantic -> update disease is now nice and silent. Thank you!

**Pull request merged**